analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs

```diff
@@ -415,6 +415,7 @@ internal sealed class KnownType
         internal static readonly KnownType System_SystemException = new KnownType("System.SystemException");
         internal static readonly KnownType System_Text_RegularExpressions_Regex = new KnownType("System.Text.RegularExpressions.Regex");
         internal static readonly KnownType System_Text_StringBuilder = new KnownType("System.Text.StringBuilder");
+        internal static readonly KnownType System_Threading_ReaderWriterLockSlim = new KnownType("System.Threading.ReaderWriterLockSlim");
```

Comment:
_Reminder:_ This will make sense after rebase on #5361 
```suggestion
        internal static readonly KnownType System_Threading_ReaderWriterLockSlim = new("System.Threading.ReaderWriterLockSlim");
```